geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -209,9 +209,10 @@ object FilterHelper extends LazyLogging {
 
       // Note: although not technically required, all known spatial predicates are also binary spatial operators
       case f: BinarySpatialOperator if isSpatialFilter(f) =>
+        if (attribute == null) logger.warn("Geometry attribute is not set.  Will include all geometries from filter")
```

Comment:
I guess another thing you could do would be:
`FilterHelper.propertyNames(filter, null).flatMap(FilterHelper.extractGeometries(filter, _, false))`
that would actually avoid having to change this method...